TAO/TAO_IDL/driver/drv_preproc.cpp

```diff
@@ -138,22 +140,68 @@ DRV_cpp_putarg (const char *str)
       throw Bailout ();
     }
 
-  if (str && ACE_OS::strchr (str, ' ') && !ACE_OS::strchr (str, '"'))
+  char *replace = nullptr;
+  if (str)
     {
-      ACE_TCHAR *buf = nullptr;
-      ACE_NEW_NORETURN (buf, ACE_TCHAR[ACE_OS::strlen (str) + 3]);
-      if (buf)
+      const char *const first_quote = ACE_OS::strchr (str, '"');
+      bool allocate_error = false;
+
+      if (ACE_OS::strchr (str, ' ') && !first_quote)
+        {
+          ACE_NEW_NORETURN (replace, char[ACE_OS::strlen (str) + 3]);
+          allocate_error = !replace;
+          if (replace)
+            {
+              replace[0] = '"';
+              ACE_OS::strcpy (replace + 1, str);
+              ACE_OS::strcat (replace, "\"");
+            }
+        }
+#ifdef ACE_WIN32
+      else if (first_quote)
+        {
+          // Escape Doublequotes on Windows
+
+          size_t quote_count = 0;
+          for (const char *quote = first_quote; quote; quote = ACE_OS::strchr (quote, '"'))
+            {
+              ++quote_count;
+              ++quote;
+            }
+
+          ACE_NEW_NORETURN (replace, char[ACE_OS::strlen (str) + quote_count + 1]);
+          allocate_error = !replace;
+          if (replace)
+            {
+              char *to = replace;
+              for (const char *from = str; *from; ++from)
+                {
+                  if (*from == '"')
+                    {
+                      *to = '\\';
+                      ++to;
+                    }
+                  *to = *from;
+                  ++to;
+                }
+              *to = '\0';
+            }
+        }
+#endif
+
+      if (allocate_error)
         {
-          buf[0] = ACE_TEXT ('"');
-          ACE_OS::strcpy (buf + 1, ACE_TEXT_CHAR_TO_TCHAR (str));
-          ACE_OS::strcat (buf, ACE_TEXT ("\""));
-          DRV_arglist[DRV_argcount++] = buf;
+          idl_global->err()->misc_error ("DRV_cpp_putarg failed to allocate memory for argument!");
+          throw Bailout ();
         }
     }
-  else
+
+  DRV_arglist[DRV_argcount++] = ACE::strnew (ACE_TEXT_CHAR_TO_TCHAR (replace ? replace : str));
+
+  if (replace)
     {
-      DRV_arglist[DRV_argcount++] =
-        ACE::strnew (ACE_TEXT_CHAR_TO_TCHAR (str));
+      delete replace;
```

Comment:
New coverity error, shouldn't this be delete[] @iguessthislldo

Reply:
Yes: #1633 and #1634.